guides/prefetching.md

```diff
@@ -122,6 +122,117 @@ The `@xdn/vue` package provides a `Prefetch` component that you can wrap around
+
+
+### Deep Fetching for HTML documents (XDN Starter)
+
+To deep fetch HTML documents, use objects that match the [DeepFetchHtmlConfig interface](https://developer.moovweb.com/docs/api/prefetch/interfaces/_sw_deepfetchplugin_.deepfetchhtmlconfig.html) and describe which HTML elements need to be prefetched via CSS selectors.
+
+For example, imagine you're configuring prefetching for a product page and you want to ensure the main product image is prefetched so that it appears immediately when the page loads. If the main product image is displayed with an HTML `img` element with a CSS class called `product-featured-media`, it can be prefetched by adding the following to the DeepFetchPlugin:
+
+```js
+import { Prefetcher } from '@xdn/prefetch/sw'
+import DeepFetchPlugin from '@xdn/prefetch/sw/DeepFetchPlugin'
+
+new Prefetcher({
+  plugins: [
+    new DeepFetchPlugin([
+      {
+        selector: 'img.product-featured-media', // CSS selector syntax - just like you would use with document.querySelector()
+        maxMatches: 1, // limits the number of matched elements to prefetch to 1 per page
+        attribute: 'src', // the attribute holding the URL to prefetching
+        as: 'image', // the type of asset being prefetched
+      },
+    ]),
+  ],
+})
+```
+
+#### Computing the URL to be prefetched (XDN Starter)
+
+In the example above the `img` element's `src` attribute contains URL that needs to be prefetched. Sometimes finding the URL to prefetch is not so straightforward. For example, apps sometimes use JavaScript to compute the URL for responsive images based on the user's device size. In such cases you can provide a `callback` function which will be passed all matching elements and decide what URLs to prefetch. Here is an example:
+
+```typescript
+import { Prefetcher, prefetch } from '@xdn/prefetch/sw'
+import DeepFetchPlugin, { DeepFetchCallbackParam } from '@xdn/prefetch/sw/DeepFetchPlugin'
+
+new Prefetcher({
+  plugins: [
+    new DeepFetchPlugin([
+      {
+        selector: 'img.grid-view-item__image',
+        maxMatches: 4,
+        as: 'image',
+        callback: deepFetchResponsiveImages,
+      },
+    ]),
+  ],
+})
+
+function deepFetchResponsiveImages({ $el, el, $ }: DeepFetchCallbackParam) {
+  const urlTemplate = $el.attr('data-src')
+  const dataWidths = $el.attr('data-widths')
+
+  if (dataWidths && urlTemplate) {
+    const widths = JSON.parse(dataWidths)
+
+    for (let width of widths.slice(0, 2)) {
+      const url = urlTemplate?.replace(/\{width\}/, width)
+      prefetch(url, 'image')
+    }
+  }
+}
+```
+
+### Deep fetching URLs in JSON responses (XDN Headless)
```

Comment:
I think this is the first time I've seen the term "XDN Headless". I suggest just removing it rather than introducing a new term.

```suggestion
### Deep fetching URLs in JSON responses
```